### ISSUE TITLE
How to export serving model with tf.lookup op

### ISSUE THREAD

**Issue by Goofy-G** [OpenNMT/OpenNMT-tf#704]
Hi
    I have questions about exporting serving model on tf 2.x. I ought to want to export serving model while lookup operation in model's call method. But I got this fall "Tried to export a function which references untracked object Tensor". I find out lookup op in openNMT-tf 's sequenceTagger model and I don't get how you can run this successfully. Could you give me some suggestion about it.

**Comment by guillaumekln** [comment]
What is your TensorFlow version?

**Comment by Goofy-G** [comment]
2.2

**Comment by guillaumekln** [comment]
How did you determine that lookup ops are causing the error? Can you post some information to reproduce the error? Command line, options, configurations, etc.

**Comment by Goofy-G** [comment]
@guillaumekln 
   This is because only using lookup op to transfer truely token from logits, if I directly output the logits instead of doing this operation, the programming 's running is ok

**Comment by guillaumekln** [comment]
Please post information to reproduce the error. We have some tests that export models with lookup ops, so it should work.

**Comment by Goofy-G** [comment]
Hi guillaumekln 
    This is my customed seqTagger code. You can directly run test/runn_test.py/test_export by pycharm. It obtianed this fall.
[TestTagger1.zip](https://github.com/OpenNMT/OpenNMT-tf/files/5027446/TestTagger1.zip)

**Comment by guillaumekln** [comment]
This question is better asked on StackOverflow as it is not directly related to OpenNMT-tf.

You get this error because the lookup table is not attached to any layers. You should store your `Tokenizer` object as an attribute of the layer.

**Comment by Goofy-G** [comment]
Thanks, bro. It has annoyed me long time

**Comment by Goofy-G** [comment]
@guillaumekln  
   When I try to export serving model, I got these warning. Whether it means being failed to load model 
   `
W0805 18:51:16.709239 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer.iter
W0805 18:51:16.709383 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer.decay
W0805 18:51:16.709438 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer.learning_rate
W0805 18:51:16.709484 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer.momentum
W0805 18:51:16.709528 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer.rho
W0805 18:51:16.709577 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.word_embedding_layer.embeddings
W0805 18:51:16.709619 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.char_embedding_layer.embeddings
W0805 18:51:16.709661 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.char_conv1D_layer.kernel
W0805 18:51:16.709706 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.char_conv1D_layer.bias
W0805 18:51:16.709748 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.dense_layer.kernel
W0805 18:51:16.709789 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.dense_layer.bias
W0805 18:51:16.709831 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.bidirectional_layer.forward_layer.cell.kernel
W0805 18:51:16.709875 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.bidirectional_layer.forward_layer.cell.recurrent_kernel
W0805 18:51:16.709916 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.bidirectional_layer.forward_layer.cell.bias
W0805 18:51:16.709958 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.bidirectional_layer.backward_layer.cell.kernel
W0805 18:51:16.710000 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.bidirectional_layer.backward_layer.cell.recurrent_kernel
W0805 18:51:16.710042 140734750109120 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'rms' for (root).model.bidirectional_layer.backward_layer.cell.bias
W0805 18:51:16.710083 140734750109120 util.py:152] A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/guide/checkpoint#loading_mechanics for details.

`

**Comment by spate141** [comment]
@Goofy-G I'm kind of trying to solve something related. Have you figure out a way to export model that is using this StaticHashTables? 

**Related issue:** 
- https://github.com/tensorflow/serving/issues/1719
